### PR TITLE
Task/189 extract cache class

### PR DESCRIPTION
**Pull request by DougMidgley**
# PR details

## What is the purpose of this pull request? (put an "X" next to an item)

_Please delete options that are not relevant._

- [ ] Documentation update
- [ ] Bug fix
- [ ] New metadata support
- [ ] Enhanced metadata
- [ ] Add a CLI option
- [ ] Add something to the core
- [X] Other, please explain: Extraction of caching logic

## What changes did you make? (Give an overview)

Moved all caching into another file to extract from main logic

## Is there anything you'd like reviewers to focus on?
MERGE 188 FIRST

## Checklist

- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] ESLint & Prettier are not outputting errors or warnings
- [x] README.md updated (if applicable)
- [ ] CHANGELOG.md updated
- [x] ran `npm run docs` to update developer docs

### CODE REVIEW

File: lib/metadataTypes/EmailSendDefinition.js [hunk 1]
```diff
@@ -147,7 +144,7 @@ class EmailSendDefinition extends MetadataType {
             // get List (required)
             if (sdl.r__list_PathName) {
                 sdl.List = {
-                    ID: Util.getListObjectIdFromCache(this.cache, sdl.r__list_PathName, 'ID'),
+                    ID: cache.getListObjectIdFromCache(sdl.r__list_PathName, 'ID'),
```
Comment (JoernBerkefeld) [inline lib/metadataTypes/EmailSendDefinition.js:147]: this method wasn't moved to Cache class but has or else this fails

- [ ] move getListObjectIdFromCache from util.js to cache.js

File: lib/metadataTypes/ImportFile.js [hunk 2]
```diff
@@ -137,8 +136,7 @@ class ImportFile extends MetadataType {
             }
         } else if (metadata.c__destinationType === 'List') {
             try {
-                metadata.destinationObjectId = Util.getListObjectIdFromCache(
-                    this.cache,
+                metadata.destinationObjectId = cache.getListObjectIdFromCache(
```
Comment (JoernBerkefeld) [inline lib/metadataTypes/ImportFile.js:139]: method not moved

- [ ] move getListObjectIdFromCache

File: lib/metadataTypes/MetadataType.js [hunk 3]
```diff
@@ -407,7 +391,8 @@ class MetadataType {
         try {
             this.removeNotCreateableFields(metadataEntry);
             const response = await this.client.soap.create(
-                this.definition.type.charAt(0).toUpperCase() + this.definition.type.slice(1),
+                overrideType ||
```
Comment (JoernBerkefeld) [inline lib/metadataTypes/MetadataType.js:394]: double-check why this was added

File: lib/metadataTypes/MetadataType.js [hunk 4]
```diff
@@ -480,7 +465,8 @@ class MetadataType {
         try {
             this.removeNotUpdateableFields(metadataEntry);
             const response = await this.client.soap.update(
-                this.definition.type.charAt(0).toUpperCase() + this.definition.type.slice(1),
+                overrideType ||
```
Comment (JoernBerkefeld) [inline lib/metadataTypes/MetadataType.js:468]: double-check why this was added v2

File: lib/util/devops.js [hunk 5]
```diff
@@ -2,7 +2,7 @@ const File = require('./file');
 const path = require('path');
 const inquirer = require('inquirer');
 const Util = require('./util');
-const git = require('simple-git/promise')();
+const git = require('simple-git')();
```
Comment (JoernBerkefeld) [inline lib/util/devops.js:5]: why change how we use simple-git?

File: lib/util/init.git.js [hunk 6]
```diff
@@ -4,7 +4,7 @@ const File = require('./file');
 const inquirer = require('inquirer');
 const Util = require('./util');
 const commandExists = require('command-exists');
-const git = require('simple-git/promise')();
+const git = require('simple-git')();
```
Comment (JoernBerkefeld) [inline lib/util/init.git.js:7]: why change how we use simple-git? v2

**Review by JoernBerkefeld — CHANGES_REQUESTED** [review]
few things left and right

File: lib/util/init.git.js (hunk 6 above)
Comment (DougMidgley) [reply lib/util/init.git.js:7]: it by default supports promises now :)

File: lib/util/devops.js (hunk 5 above)
Comment (DougMidgley) [reply lib/util/devops.js:5]: as before ;) by default supports promises now

**Review by JoernBerkefeld — APPROVED** [review]

**Pull request merged**